onnx/defs/tensor/defs.cc

```diff
@@ -474,16 +477,37 @@ ONNX_OPERATOR_SET_SCHEMA(
           if (!hasNInputShapes(ctx, 1)) {
             return;
           }
-
+          auto input_type = ctx.getInputType(0);
+          if (!input_type->has_tensor_type()) {
+              throw InferenceError("Input 0 expected to have tensor type");
+          }
+          const TensorShapeProto& shape = input_type->tensor_type().shape();
           std::vector<int64_t> perm;
-          if (!getRepeatedAttribute(ctx, "perm", perm)) {
-            for (int i =
-                     ctx.getInputType(0)->tensor_type().shape().dim_size() - 1;
-                 i >= 0;
-                 --i) {
+          bool has_perm_attr = getRepeatedAttribute(ctx, "perm", perm);
+          if (!has_perm_attr) {
+            for (int i = shape.dim_size() - 1; i >= 0; --i) {
               perm.push_back(i);
             }
           }
+          else if(!perm.empty()){
+              for (int64_t fromDimIndex : perm)
+                  if (shape.dim_size() <= fromDimIndex || fromDimIndex >= std::numeric_limits<int>::max()) {
```

Comment:
if (! ((0 <= fromDimIndex) && (fromDimIndex < shape.dim_size()))

Reply:
if (fromDimIndex < 0 || shape.dim_size() <= fromDimIndex || fromDimIndex >= std::numeric_limits<int>::max())  ?

Reply:
Thanks. Will change it.